content/en/blog/2021/revision-tags/index.md

```diff
@@ -0,0 +1,111 @@
+
+{{< image link="./tags-updated.png" caption="Updated revision tags" >}}
+
+Once you're satisfied with the upgrade to the new control-plane revision, you can remove the old control plane.
+
+## Stable revision tags in action
+
+To create a new `prod-stable` tag for a revision `1-9-5`, run the following command:
+
+{{< text bash >}}
+$ istioctl x revision tag set prod-stable --revision 1-9-5
+{{< /text >}}
+
+You can then label your namespaces with the `istio.io/rev=prod-stable` label. Note, if you installed a `default` revision (i.e., no revision) of Istio, you will first have to remove the standard injection label:
+
+{{< text bash >}}
+$ kubectl label ns istioinaction istio-injection-
+$ kubectl label ns istioinaction istio.io/rev=prod-stable
+{{< /text >}}
+
+You can list the tags in your mesh with the following:
+
+{{< text bash >}}
+$ istioctl x revision tag list
+
+TAG         REVISION NAMESPACES
+prod-stable 1-9-5
+{{< /text >}}
+
+A tag is implemented with a `MutatingWebhookConfiguration`. You can verify a corresponding `MutatingWebhookConfiguration` has been created:
+
+{{< text bash >}}
+$ kubectl get MutatingWebhookConfiguration
+
+NAME                             WEBHOOKS   AGE
+istio-revision-tag-prod-stable   2          75s
+istio-sidecar-injector           1          5m32s
+{{< /text >}}
+
+Let's say you are trying to canary a new revision of the control plane based on 1.10.0. First you would install the new version using a revision:
+
+{{< text bash >}}
+$ istioctl install -y --set profile=minimal --revision 1-10-0
+{{< /text >}}
+
+You can create a new tag called `prod-canary` and point that to your `1-10-0` revision:
+
+{{< text bash >}}
+$ istioctl x revision tag set prod-canary --revision 1-10-0
+{{< /text >}}
+
+Then label your namespaces accordingly:
+
+{{< text bash >}}
+$ kubectl label ns istioinaction-canary istio.io/rev=prod-canary
+{{< /text >}}
+
+If you list out the tags in your mesh, you will see two stable tags pointing to two different revisions:
+
+{{< text bash >}}
+$ istioctl x revision tag list
+
+TAG         REVISION NAMESPACES
+prod-stable 1-9-5    istioinaction
+prod-canary 1-10-0   istioinaction-canary
+{{< /text >}}
+
+Any of the namespaces that you have labeled with `istio.io/rev=prod-canary` will be injected by the control plane that corresponds to the `prod-canary` stable tag name (which in this example points to the `1-10-0` revision). When you're ready, you can switch the `prod-stable` tag to the new control plane with:
+
+{{< text bash >}}
+$ istioctl x revision tag set prod-stable --revision 1-10-0 --overwrite
+{{< /text >}}
+
+Any time you switch a tag to point to a new revision, you should restart the workloads in any respective namespace to pick up the new revision's proxy.
+
+When both the `prod-stable` and `prod-canary` tags both point to the same new revision, it may be safe to remove the old revision like the following:
```

Comment:
Would it be better to say When both the `prod-stable` and `prod-canary` tags no longer point to the old version, ...  Or maybe When no revisions or revision tags point to the old version,... ?

Reply:
Yes, user may have many other tags - "when no tags point to a revision" is better.

May also note somewhere that 'Users should make sure the revision is not directly used by a namespace 
or object' ( since in 1.10 you can also set it as a label on a workload, without any label in the namespace ).